Library/Homebrew/livecheck/livecheck.rb

```diff
@@ -0,0 +1,384 @@
+
+        if latest.blank?
+          no_versions_msg = "Unable to get versions"
+          raise TypeError, no_versions_msg unless args.json?
+
+          next version_info if version_info.is_a?(Hash) && version_info["status"] && version_info["messages"]
+
+          next status_hash(formula, "error", [no_versions_msg], args: args)
+        end
+
+        if (m = latest.to_s.match(/(.*)-release$/)) && !current.to_s.match(/.*-release$/)
+          latest = Version.new(m[1])
+        end
+
+        info = {
+          "formula" => formula_name(formula, args: args),
+          "version" => {
+            "current"             => current.to_s,
+            "latest"              => latest.to_s,
+            "outdated"            => formula.stable? ? (current < latest) : (current != latest),
+            "newer_than_upstream" => formula.stable? && (current > latest),
+          },
+          "meta"    => {
+            "livecheckable" => formula.livecheckable?,
+          },
+        }
+        info["meta"]["head_only"] = !formula.stable? unless formula.stable?
+        info["meta"].merge!(version_info["meta"]) if version_info.present? && version_info.key?("meta")
+
+        next if args.newer_only? && !info["version"]["outdated"]
+
+        has_a_newer_upstream_version ||= true
+
+        if args.json?
+          info.except!("meta") unless args.verbose?
+          next info
+        end
+
+        print_latest_version(info, args: args)
+        nil
+      rescue => e
+        Homebrew.failed = true
+
+        if args.json?
+          status_hash(formula, "error", [e.to_s], args: args)
+        elsif !args.quiet?
+          onoe "#{Tty.blue}#{formula_name(formula, args: args)}#{Tty.reset}: #{e}"
+          nil
+        end
+      end
+
+      if [args.newer_only?, !has_a_newer_upstream_version, !args.json?, !args.debug?].all?
+        puts "No newer upstream versions."
+      end
+
+      puts JSON.generate(formulae_checked.compact) if args.json?
+    end
+
+    def formula_name(formula, args:)
+      args.full_name? ? formula.full_name : formula.name
+    end
+
+    def status_hash(formula, status_str, messages = nil, args:)
+      status_hash = {
+        "formula" => formula_name(formula, args: args),
+        "status"  => status_str,
+      }
+      status_hash["messages"] = messages.presence
+
+      if args.verbose?
+        status_hash["meta"] = {
+          "livecheckable" => formula.livecheckable?,
+        }
+        status_hash["meta"]["head_only"] = !formula.stable? unless formula.stable?
+      end
+
+      status_hash
+    end
+
+    def skip_conditions(formula, args:)
+      if formula.deprecated? && !formula.livecheckable?
+        return status_hash(formula, "deprecated", args: args) if args.json?
+
+        unless args.quiet?
+          puts "#{Tty.red}#{formula_name(formula, args: args)}#{Tty.reset} : deprecated"
+          return
+        end
+      end
+
+      if formula.to_s.include?("@") && !formula.livecheckable?
```

Comment:
```suggestion
      if formula.versioned_formula? && !formula.livecheckable?
```

Might be worth reading the `Formula` class in general to find any other cases that could be used like this.